src/collision.cpp

```diff
@@ -523,30 +523,34 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 				speed_f->X *= bounce;
 			} else {
 				speed_f->X = 0;
-				accel_f.X = 0;
 			}
 			result.collides = true;
 		} else if (nearest_collided == COLLISION_AXIS_Y) {
-			if(bounce < -1e-4 && fabsf(speed_f->Y) > BS * 3) {
+			if (bounce < -1e-4 && fabsf(speed_f->Y) > BS * 3) {
 				speed_f->Y *= bounce;
 			} else {
+				if (speed_f->Y < 0.0f) {
+					// FIXME: This code is not necessary but results in a more responsive experience
+					// at high dtime (e.g. entities). `collisionMoveSimple` currently needs two calls
+					// to line up with the colliding face well enough to trigger the ground detection.
+					result.touching_ground = true;
```

Comment:
well either of them is necessary or `touching_ground` would never be true. and I think this place makes more sense.

Reply:
I agree. The collision code does work well in the current state so that the bottom loop can be optimized such that only `step_up` nodes are processed there.
Shall I reword the comment here? Please provide a suggestion.